isaaclab_arena/embodiments/agibot/agibot.py

```diff
@@ -0,0 +1,262 @@
+from isaaclab_arena.embodiments.embodiment_base import EmbodimentBase
+from isaaclab_arena.embodiments.franka.franka import FrankaMimicEnv
+from isaaclab_arena.utils.pose import Pose
+
+
+@register_asset
+class AgibotEmbodiment(EmbodimentBase):
+    """Embodiment for the Agibot robot."""
+
+    name = "agibot"
+    default_mimic_arm_mode = MimicArmMode.LEFT
+
+    def __init__(
+        self,
+        enable_cameras: bool = False,
+        initial_pose: Pose | None = None,
+        mimic_arm_mode: MimicArmMode | None = None,
+    ):
+        super().__init__(enable_cameras, initial_pose)
+        self.mimic_arm_mode = mimic_arm_mode or self.default_mimic_arm_mode
+        self.scene_config = (
+            AgibotLeftArmSceneCfg() if self.mimic_arm_mode == MimicArmMode.LEFT else AgibotRightArmSceneCfg()
+        )
+        self.action_config = (
+            AgibotLeftArmActionsCfg() if self.mimic_arm_mode == MimicArmMode.LEFT else AgibotRightArmActionsCfg()
+        )
+        self.observation_config = AgibotObservationsCfg()
+        self.mimic_env = AgibotMimicEnv
+
+
+@configclass
+class AgibotLeftArmSceneCfg:
+    """Scene configuration for the Agibot left arm."""
+
+    robot = AGIBOT_A2D_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
+
+    ee_frame = FrameTransformerCfg(
+        prim_path="{ENV_REGEX_NS}/Robot/base_link",
+        debug_vis=False,
+        target_frames=[
+            FrameTransformerCfg.FrameCfg(
+                prim_path="{ENV_REGEX_NS}/Robot/gripper_center",
+                name="left_end_effector",
+                offset=OffsetCfg(
+                    pos=[0.0, 0.0, 0.0],
+                    rot=[
+                        0.7071,
+                        0.0,
+                        -0.7071,
+                        0.0,
+                    ],  # rpy: [0, -90, 0] to make the gripper direction from +Z to +X
+                ),
+            ),
+        ],
+    )
+
+    def __post_init__(self):
+        # Add a marker to the end-effector frame
+        marker_cfg = FRAME_MARKER_CFG.copy()
+        marker_cfg.markers["frame"].scale = (0.1, 0.1, 0.1)
+        marker_cfg.prim_path = "/Visuals/FrameTransformer"
+        self.ee_frame.visualizer_cfg = marker_cfg
+
+
+@configclass
+class AgibotRightArmSceneCfg(AgibotLeftArmSceneCfg):
+    """Scene configuration for the Agibot right arm."""
+
+    def __post_init__(self):
+        # update the ee_frame for the right arm
+        self.ee_frame = FrameTransformerCfg(
+            prim_path="{ENV_REGEX_NS}/Robot/base_link",
+            debug_vis=False,
+            target_frames=[
+                FrameTransformerCfg.FrameCfg(
+                    prim_path="{ENV_REGEX_NS}/Robot/right_gripper_center",
+                    name="right_end_effector",
+                    offset=OffsetCfg(
+                        pos=[0.0, 0.0, 0.0],
+                    ),
+                ),
+            ],
+        )
```

Comment:
This is a bit of a strange use of inheritance. Suggestion to instead create a function which takes in `arm_mode` and constructs a `AgibotSceneCfg` appropriately.

Maybe:

```
@configclass
class AgibotSceneCfg:
    robot = AGIBOT_A2D_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
    arm_mode: ArmMode
    eef_frame: MISSING

   def __post_init__(self):
        if self.arm_mode == ArmMode.LEFT:
               eef = ...
```

Then above in `AgibotEmbodiment` you can do:

```
scene_config = AgibotSceneCfg(arm_mode=ArmMode.LEFT)
```

And we don't need these two classes with inheritance etc.